vmm/src/memory_manager.rs

```diff
@@ -716,12 +721,18 @@ impl MemoryManager {
                                 * virtio_devices::VIRTIO_MEM_ALIGN_SIZE,
                         );
 
+                        // When `prefault` is set by vm_restore, memory manager
+                        // will create ram region with `prefault` option in
+                        // restore config rather than same option in zone
                         let region = MemoryManager::create_ram_region(
                             &None,
                             0,
                             start_addr,
                             hotplug_size as usize,
-                            false,
+                            match prefault {
+                                Some(pf) => pf,
+                                None => zone.prefault,
+                            },
```

Comment:
Here you're modifying the existing behavior, by allowing the user to prefault the virtio-mem region. I'm concerned about this one as this means the entire region will be prefaulted, even if nothing's plugged yet. Imagine a virtio-mem region of 64G that a user would create only for the purpose of having it around, but with no guarantee he's gonna use it. In this case, if the user wants to enable prefault, the full 64G will be populated.
For this specific use case, I'm not sure we should do the prefaulting.
@rbradford I'd like your feedback on this.